### ISSUE TITLE
Linux specific bug after creating channel  

### ISSUE THREAD

**Issue by stridentbean** [lightningnetwork/lnd#463]
I'm running BTCD and LND in testnet on an old linux box. Everything had been running fine for a month or so. Then I updated master this week and see the following error. This happened after I created a new channel. It booted up fine without any channels. I send a few on chain transactions too without issue. 

I think it has something to do with this commit. https://github.com/lightningnetwork/lnd/commit/de3af9b0c01799cc0cb940ed6fed0bb94b71a3ff

Relevant logs: 

`2017-12-07 22:01:09.291 [INF] NTFN: New spend subscription: utxo=cedf18dbccccdf3ad0d26f7d0b6f06964a879b57268fa2fd277beb79ae404adb:0
2017-12-07 22:01:09.292 [INF] PEER: peerID(2) loading ChannelPoint(cedf18dbccccdf3ad0d26f7d0b6f06964a879b57268fa2fd277beb79ae404adb:0)
2017-12-07 22:01:09.293 [INF] LNWL: Close observer for ChannelPoint(cedf18dbccccdf3ad0d26f7d0b6f06964a879b57268fa2fd277beb79ae404adb:0) active
2017-12-07 22:01:09.293 [INF] BRAR: ChannelPoint(cedf18dbccccdf3ad0d26f7d0b6f06964a879b57268fa2fd277beb79ae404adb:0) is now live, abandoning state contract for live version
2017-12-07 22:01:09.293 [INF] NTFN: New block epoch subscription
2017-12-07 22:01:09.293 [INF] NTFN: Cancelling spend notification for out_point=cedf18dbccccdf3ad0d26f7d0b6f06964a879b57268fa2fd277beb79ae404adb:0, spend_id=1
2017-12-07 22:01:09.300 [INF] HSWC: ChannelLink(cedf18dbccccdf3ad0d26f7d0b6f06964a879b57268fa2fd277beb79ae404adb:0) is starting
2017-12-07 22:01:09.300 [INF] HSWC: Added channel link with chan_id=db4a40ae79eb7b27fda28f26579b874a96066f0b7d6fd2d03adfccccdb18dfce, short_chan_id=((lnwire.ShortChannelID) {
 BlockHeight: (uint32) 1254015,
 TxIndex: (uint32) 47,
 TxPosition: (uint16) 0
}
)
2017-12-07 22:01:09.300 [INF] SRVR: Established connection to: 195.181.243.196:9735
2017-12-07 22:01:09.300 [INF] HSWC: ChannelLink(cedf18dbccccdf3ad0d26f7d0b6f06964a879b57268fa2fd277beb79ae404adb:0) has exited
2017-12-07 22:01:09.301 [INF] PEER: unable to read message from 195.181.243.196:9735: read tcp 192.168.2.2:57610->195.181.243.196:9735: use of closed network connection
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x4 pc=0x13330]
`

`goroutine 222 [running]:
sync/atomic.loadUint64(0x112ae3c4, 0x0, 0x1127a238)
        /usr/local/go/src/sync/atomic/64bit_arm.go:10 +0x3c
github.com/lightningnetwork/lnd/htlcswitch.(*packetQueue).TotalHtlcAmount(0x112ae3c0, 0x564dc0, 0x0)
        /home/pi/gocode/src/github.com/lightningnetwork/lnd/htlcswitch/queue.go:197 +0x24
github.com/lightningnetwork/lnd/htlcswitch.(*channelLink).Bandwidth(0x11314820, 0x11632024, 0x11314820)
        /home/pi/gocode/src/github.com/lightningnetwork/lnd/htlcswitch/link.go:1058 +0x40
github.com/lightningnetwork/lnd/htlcswitch.(*channelLink).htlcManager(0x11314820)
        /home/pi/gocode/src/github.com/lightningnetwork/lnd/htlcswitch/link.go:362 +0x5c
created by github.com/lightningnetwork/lnd/htlcswitch.(*channelLink).Start
        /home/pi/gocode/src/github.com/lightningnetwork/lnd/htlcswitch/link.go:267 +0x108`

**Comment by Roasbeef** [comment]
Ah I know what it is. It's due to alignment on ARM systems. Usage of atomics requires that the integers used be byte aligned.

**Comment by halseth** [comment]
Wow, today I learned! Nice find 😀

**Comment by stridentbean** [comment]
Thanks, guys!

**Comment by Roasbeef** [comment]
Before: 
```
⛰i  structlayout github.com/lightningnetwork/lnd/htlcswitch packetQueue
packetQueue.queueLen int32: 0-4 (size 4, align 4)
padding: 4-8 (size 4, align 0)
packetQueue.totalHtlcAmt int64: 8-16 (size 8, align 8)
packetQueue.queueCond *sync.Cond: 16-24 (size 8, align 8)
packetQueue.queueMtx.state int32: 24-28 (size 4, align 4)
packetQueue.queueMtx.sema uint32: 28-32 (size 4, align 4)
packetQueue.queue []*github.com/lightningnetwork/lnd/htlcswitch.htlcPacket: 32-56 (size 24, align 8)
packetQueue.outgoingPkts chan *github.com/lightningnetwork/lnd/htlcswitch.htlcPacket: 56-64 (size 8, align 8)
packetQueue.freeSlots chan struct{}: 64-72 (size 8, align 8)
packetQueue.wg.noCopy sync.noCopy: 72-72 (size 0, align 1)
packetQueue.wg.state1 [12]byte: 72-84 (size 12, align 1)
packetQueue.wg.sema uint32: 84-88 (size 4, align 4)
packetQueue.quit chan struct{}: 88-96 (size 8, align 8)
```

After: 
```
⛰i  structlayout  -json github.com/lightningnetwork/lnd/htlcswitch packetQueue | structlayout-optimize
packetQueue.queue []*github.com/lightningnetwork/lnd/htlcswitch.htlcPacket: 0-24 (size 24, align 8)
packetQueue.wg struct: 24-40 (size 16, align 8)
packetQueue.freeSlots chan struct{}: 40-48 (size 8, align 8)
packetQueue.queueCond *sync.Cond: 48-56 (size 8, align 8)
packetQueue.queueMtx struct: 56-64 (size 8, align 8)
packetQueue.outgoingPkts chan *github.com/lightningnetwork/lnd/htlcswitch.htlcPacket: 64-72 (size 8, align 8)
packetQueue.totalHtlcAmt int64: 72-80 (size 8, align 8)
packetQueue.quit chan struct{}: 80-88 (size 8, align 8)
packetQueue.queueLen int32: 88-92 (size 4, align 8)
padding: 92-96 (size 4, align 0)
```

**Comment by creslinux** [comment]
Builds on Raspberry Pi3 also falling foul of this bug. 
The Pi3 is 64bit, but only the Pi2 32bit GO binaries are provided in standard builds of its Raspbian OS. 
```

Stack trace: 
2017-12-18 00:53:10.576 [INF] HSWC: ChannelLink(105b920b4a61f1bbee93cf1c37f756a870f9b13bb4df60cd21cfa0c7abca6534:0) has exited
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x4 pc=0x13330]

goroutine 246 [running]:
sync/atomic.loadUint64(0x11399ac4, 0x0, 0x1117e714)
	/usr/local/go/src/sync/atomic/64bit_arm.go:10 +0x3c
github.com/lightningnetwork/lnd/htlcswitch.(*packetQueue).TotalHtlcAmount(0x11399ac0, 0x2faf080, 0x0)
	/home/btcn/gocode/src/github.com/lightningnetwork/lnd/htlcswitch/queue.go:197 +0x24
github.com/lightningnetwork/lnd/htlcswitch.(*channelLink).Bandwidth(0x112aa5a0, 0x11267e24, 0x112aa5a0)
	/home/btcn/gocode/src/github.com/lightningnetwork/lnd/htlcswitch/link.go:1060 +0x40
github.com/lightningnetwork/lnd/htlcswitch.(*channelLink).htlcManager(0x112aa5a0)
	/home/btcn/gocode/src/github.com/lightningnetwork/lnd/htlcswitch/link.go:362 +0x5c
created by github.com/lightningnetwork/lnd/htlcswitch.(*channelLink).Start
	/home/btcn/gocode/src/github.com/lightningnetwork/lnd/htlcswitch/link.go:267 +0x108

```

**Comment by Roasbeef** [comment]
This was fixed by #507.
